src/cpu/intel.rs

```diff
@@ -201,40 +202,67 @@ pub fn read_power_limit(file_path: String) -> f64 {
 }
 
 pub fn get_intel_cpu_cunter(results: &mut HashMap<String, f64>) {
+    // --- Read using Power Capping Framework (Linux only) ---
     // let zone = setup_rapl_data();
-    let start_time = Instant::now();
-    let mut prev_time = Instant::now();
-    let mut now = Instant::now();
-
+    // let start_time = Instant::now();
+    // let mut prev_time = Instant::now();
+    // let mut now = Instant::now();
     //    let mut data =
     //      calculate_power_metrics(zone.get(0).unwrap().to_owned(), now, start_time, prev_time);
 
     unsafe {
+        // --- Read using MSR ---
+
+        // TODO: these values are constant, read them once and store them
+        // The MSR only store integer values, but they represent floating point values.
+        // The MSR_RAPL_POWER_UNIT MSR contains the units for the RAPL MSRs for a specific intel chip.
+        // it contains three units, which represent the time, power, and energy increments in the RAPL MSRs.
         let core_energy_units: u64 = read_msr_on_core(MSR_RAPL_POWER_UNIT, 0).unwrap();
-        let energy_unit: u64 = (core_energy_units & INTEL_TIME_UNIT_MASK) >> 8;
```

Comment:
Notice this used `INTEL_TIME_UNIT_MASK` instead of `INTEL_ENGERY_UNIT_MASK`. Also, the offset 8 does not match the used mask.